cfgmgr/portmgr.cpp

```diff
@@ -149,38 +118,42 @@ void PortMgr::doTask(Consumer &consumer)
                 {
                     admin_status = fvValue(i);
                 }
-                else if (fvField(i) == "learn_mode")
-                {
-                    learn_mode = fvValue(i);
-                }
-                else if (fvField(i) == "tpid")
+                else 
                 {
-                    tpid = fvValue(i);
+                    field_values.emplace_back(i);
                 }
             }
 
-            if (!mtu.empty())
+            for (auto &entry : field_values)
             {
-                setPortMtu(alias, mtu);
-                SWSS_LOG_NOTICE("Configure %s MTU to %s", alias.c_str(), mtu.c_str());
+                writeConfigToAppDb(alias, fvField(entry), fvValue(entry));
+                SWSS_LOG_NOTICE("Configure %s %s to %s", alias.c_str(), fvField(entry).c_str(), fvValue(entry).c_str());
             }
 
-            if (!admin_status.empty())
+            if (!portOk)
             {
-                setPortAdminStatus(alias, admin_status == "up");
-                SWSS_LOG_NOTICE("Configure %s admin status to %s", alias.c_str(), admin_status.c_str());
+                SWSS_LOG_INFO("Port %s is not ready, pending...", alias.c_str());
+
+                writeConfigToAppDb(alias, "mtu", mtu);
+                writeConfigToAppDb(alias, "admin_status", admin_status);
+                field_values.clear();
+                field_values.emplace_back("mtu", mtu);
+                field_values.emplace_back("admin_status", admin_status);
+                it->second = KeyOpFieldsValuesTuple{alias, SET_COMMAND, field_values};
```

Comment:
Can you confirm if in the retry process, any other fields (learnmode or mtu) are modified by the user, it gets reflected?

Reply:
Yes, this is covered by unit test case "ConfigureDuringRetry".